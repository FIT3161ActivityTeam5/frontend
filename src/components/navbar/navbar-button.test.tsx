import React from 'react';
import renderer from 'react-test-renderer';

import NavbarButton from './navbar-button';

describe('<NavbarButton />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <NavbarButton title="Confirm" icon="Icon" active />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when inactive', () => {
    const tree = renderer.create(
      <NavbarButton title="Confirm" icon="Icon" active={false} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
