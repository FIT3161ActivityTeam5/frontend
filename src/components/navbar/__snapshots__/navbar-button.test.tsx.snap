// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "opacity": 1,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
    }
  }
>
  Icon
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins_500Medium",
        },
        Object {
          "color": "rgba(139, 92, 246, 1)",
          "fontSize": 12,
          "lineHeight": 16,
        },
      ]
    }
  >
    Confirm
  </Text>
</View>
`;

exports[`<NavbarButton /> renders correctly when inactive 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "opacity": 1,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
    }
  }
>
  Icon
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins_500Medium",
        },
        Object {
          "color": "rgba(156, 163, 175, 1)",
          "fontSize": 12,
          "lineHeight": 16,
        },
      ]
    }
  >
    Confirm
  </Text>
</View>
`;
